### PR TITLE
✨ [New Feature]: #134 インラインイメージを1トークンとして扱う

### LINKED ISSUE #134: [Phase 3] インラインイメージ (BI/ID/EI) のトークン化
#### 概要

コンテンツストリーム中のインラインイメージ構文 `BI ... ID <bytes> EI` をトークン化する。`BI` 以降は辞書としてパースし、`ID` から `EI` までの画像バイト列はバイナリブロックとして抜き出す。

#### タスク

- [ ] `BI` を検出して以降のキー/値ペアを辞書としてパース
- [ ] `ID` 直後の改行を 1 バイトだけスキップ (PDF 仕様準拠)
- [ ] `EI` までの生バイト列を `Uint8Array` として抽出
- [ ] 取得結果を 1 個のトークン (例: `{ type: TokenType.InlineImage, dict, data, offset }`) として ContentStreamTokenizer から emit
- [ ] 不正フォーマット (`EI` が見つからない等) は `Result<..., PdfError>` で失敗を返す

#### 変更ファイル

- `packages/core/src/content-stream/tokenizer/inline-image.ts` (新規)
- `packages/core/src/content-stream/tokenizer/inline-image.test.ts` (新規、テスト対象と同フォルダ)

#### 依存関係

Blocked by: #132

### PR DESCRIPTION
**Pull request by DIO0550**
## 概要

PDF content stream 内の inline image 構文 `BI ... ID <bytes> EI` を `ContentStreamTokenizer` で 1 個の `TokenType.InlineImage` として返せるようにしました。

### 🎯 変更の種類

- [ ] 🐛 バグ修正 (Bug fix)
- [x] ✨ 新機能 (New feature)
- [ ] ♻️ リファクタリング (Refactoring)
- [ ] 🎨 UI/UX 改善 (UI/UX improvement)
- [ ] 🐎 パフォーマンス改善 (Performance improvement)
- [x] 🚨 テスト追加/修正 (Tests)
- [ ] 📖 ドキュメント更新 (Documentation)
- [ ] 🔧 設定変更 (Configuration)
- [ ] 🗑️ 削除 (Removal)

#### 追加された機能・修正

- `TokenType.InlineImage` と `TokenInlineImage` を追加
- inline image 辞書を key/value pair の順序保持配列として保持
- `ID` 以降の画像データを `Uint8Array.subarray` で保持
- `Tokenizer.seek(position)` を `Option<PdfError>` ベースで追加
- `BI` 検出時に raw byte scan で `EI` 境界を探索し、通常 tokenization に復帰
- `ID` / `EI` 欠損、dict key/value 不正、nested `BI` を `CONTENT_STREAM_INLINE_IMAGE_INVALID` として返す

#### 変更されたファイル

- `packages/core/src/content-stream/tokenizer/index.ts`
- `packages/core/src/content-stream/tokenizer/inline-image.ts`
- `packages/core/src/lexer/tokenizer/index.ts`
- `packages/core/src/pdf/types/token/index.ts`
- `packages/core/src/pdf/errors/error/index.ts`
- 関連する colocated test

#### 削除されたファイル・機能

- `BI` を `NOT_IMPLEMENTED` として拒否する挙動を削除

## 📊 システム図

```mermaid
flowchart TD
    A[ContentStreamTokenizer.nextToken] --> B{Keyword?}
    B -->|No| C[return raw token]
    B -->|Yes, not BI| D[return Operator]
    B -->|BI| E[NEW: readInlineImage]
    E --> F[Parse dict key/value until ID]
    F --> G[Scan raw bytes for boundary-delimited EI]
    G --> H[NEW: TokenType.InlineImage]
    H --> I[Tokenizer.seek EI後]
    I --> J[Next normal token]

    style E fill:#ffe4b5,stroke:#d97706,stroke-width:2px
    style H fill:#ffe4b5,stroke:#d97706,stroke-width:2px
    style I fill:#ffe4b5,stroke:#d97706,stroke-width:2px
```

## 📋 関連 Issue

- Closes #134

### テスト実行方法

```bash
pnpm run test:run
pnpm run lint
pnpm run typecheck
```

### テスト項目

- [x] 単体テスト (Unit tests)
- [x] 統合テスト (Integration tests)
- [ ] E2E テスト (End-to-end tests)
- [ ] 手動テスト (Manual testing)

### テスト結果

- `pnpm run test:run`: 111 files / 1507 tests passed
- `pnpm run lint`: passed
- `pnpm run typecheck`: passed

## 🔍 レビューポイント

- `EI` 終端判定が boundary 条件に沿っているか
- `ID` 直後の LF / CR / CRLF / space / 空白なしの扱い
- `Tokenizer.seek` の公開 API と `Option<PdfError>` の使い方
- inline image dictionary を semantic validation せず token pair として保持する方針

## ⚠️ 破壊的変更

- [ ] この変更は既存の API に破壊的変更を含みます

### 注意事項

- `pnpm run lint` 実行時、Biome が `.pnpm-store` の broken symlink warning を出しますが、lint コマンド自体は成功しています。

## ✅ チェックリスト

- [x] コードレビューの準備ができている
- [x] テストが正常に実行される
- [x] ドキュメントが更新されている（必要に応じて）
- [x] コミットメッセージが適切な形式で書かれている
- [x] 関連する Issue が PR の「Development」に Linked されている
- [x] PR 本文に `Closes #` / `Fixes #` / `Refs #` などで Issue が記載されている
- [x] セルフレビューを実施した
- [x] 破壊的変更がある場合は明記されている

### CODE REVIEW

File: packages/core/src/content-stream/tokenizer/inline-image.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+  const afterEndOffset = offset + InlineImageEnd.length;
+  if (afterEndOffset >= data.length) {
+    return true;
+  }
+
+  return (
+    isPdfWhitespace(data[afterEndOffset]) ||
+    isPdfDelimiter(data[afterEndOffset])
+  );
```
[the remark this answers is not in the document]
Comment (DIO0550) [reply packages/core/src/content-stream/tokenizer/inline-image.ts:338]: Thanks. This is a valid limitation of PDF inline images in general, but I am leaving this out of scope for #134 intentionally. The implementation plan for this issue explicitly limits tokenizer behavior to boundary-delimited EI detection and notes that complete disambiguation requires interpreting filter/length information beyond the tokenizer layer. This PR preserves that scoped behavior and has tests for the boundary cases covered by the issue.
